enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -0,0 +1,398 @@
+
+- Conversely, if the request aligns with the policy or falls within an exception due to namespace labeling, the process advances eventually adhering to Kubernetes' standard scheduling steps (or direct pod placement with spec.nodeName).
+
+8. Emergency Overrides and Adjustments:
+
+- For emergency scenarios necessitating temporary deviations from the norm, Cluster Administrators have the option to swiftly modify namespace labels or adjust the ValidatingAdmissionPolicy and ValidatingAdmissionPolicyBinding resources. This flexibility ensures that critical operations can proceed unhindered, even under exceptional circumstances.
+
+**Note:** While the ValidatingAdmissionPolicy feature is in Tech Preview within OpenShift, an initial step requires Cluster Administrators to enable the TechPreview feature gate on their cluster and subsequently restart the API Servers to facilitate the creation of policies and bindings. It's important to note that activating this feature marks the cluster as non-upgradable. This significant consideration should be carefully weighed when deciding to implement this enforcement approach.
+
+### API Extensions
+
+No new API or fields added. Solution works out of the box, given we ship necessary tolerations for control plane workloads managed by us.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Cluster admins won´t be able to taint control-plane nodes in this scenario. It would not be possible to apply this workflow.
+
+#### Standalone Clusters
+
+This workflow shines here, where cluster admins have full control over their clusters. This workflow is targeting this topology.
+
+#### Single-node Deployments or MicroShift
+
+Tainting a single-node cluster would not make sense, then this workflow is not aimed at those types of topologies.
+
+### Implementation Details/Notes/Constraints
+
+See Alternatives section. The previous discarded approach required actual implementation of a admission plugin/controller or some other way to avoid tolerations with new code provided by us, but the accepted workflow is now available out-of-the-box with ValidatingAdmissionPolicies with CEL (still in tech preview, but available if enabled) combined with the NoExecute taint enforcement. So no implementation from our side is needed, only documentation and guidance on how to apply this workflow.
+
+### Risks and Mitigations
+
+- Unintended Namespace Labeling: Administrators or namespace owners might inadvertently label namespaces in a way that bypasses the scheduling restrictions, potentially exposing control plane or special nodes to unauthorized workloads.
+  - Mitigation: Implement clear policies and documentation around the labeling of namespaces. Standardize a simple workflow, similar to the one mentioned above, so users don't deviate too much, while still allowing them to if they want.
```

Comment:
nit: do we want to mention that this mitigation is intended for customers? 
i.e
```suggestion
  - Mitigation: Administrators implement clear policies and documentation around the labeling of namespaces. Standardize a simple workflow, similar to the one mentioned above, so users don't deviate too much, while still allowing them to if they want.
```